tools/misbehaving-jmx-server/src/main/java/org/datadog/misbehavingjmxserver/App.java

```diff
@@ -0,0 +1,110 @@
+package org.datadog.misbehavingjmxserver;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.rmi.registry.LocateRegistry;
+import java.rmi.registry.Registry;
+import java.rmi.server.RMISocketFactory;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.MBeanRegistrationException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnectorServer;
+import javax.management.remote.JMXConnectorServerFactory;
+import javax.management.remote.JMXServiceURL;
+import javax.management.remote.rmi.RMIConnectorServer;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import io.javalin.*;
+import lombok.extern.slf4j.Slf4j;
+
+class AppConfig {
+    // RMI Port is used for both registry and the JMX service
+    @Parameter(names = {"--rmi-port", "-rp"})
+    public int rmiPort = 1099;
+
+    @Parameter(names = {"--rmi-interface", "-ri"})
+    public String rmiInterface = "localhost";
+
+    @Parameter(names = {"--control-port", "-cp"})
+    public int controlPort = 8080;
+}
+
+@Slf4j
+public class App 
+{
+    public static void main( String[] args ) throws IOException, MalformedObjectNameException, InstanceAlreadyExistsException, MBeanRegistrationException, NotCompliantMBeanException
+    {
+        AppConfig config = new AppConfig();
+        JCommander jCommander = JCommander.newBuilder()
+                .addObject(config)
+                .build();
+        try {
+            jCommander.parse(args);
+        } catch (Exception e) {
+            jCommander.usage();
+            System.exit(1);
+        }
+
+        InterruptibleRMISocketFactory customRMISocketFactory = new InterruptibleRMISocketFactory();
+        // I don't think this call is actually important for jmx, the below 'env' param to JMXConnectorServerFactory is the important one
```

Comment:
Test this, if its not needed, remove it, but maybe a comment leftover would be handy. The global vs jmxconnector rmi socket factory is confusing